File: src/brs/web/api/http/handler/GetDGSGood.java
package brs.web.api.http.handler;

import brs.BurstException;
import brs.services.ParameterService;
import brs.web.api.http.ApiServlet;
import brs.web.api.http.common.JSONData;
import brs.web.api.http.common.LegacyDocTag;
import com.google.gson.JsonElement;

import javax.servlet.http.HttpServletRequest;

import static brs.web.api.http.common.Parameters.GOODS_PARAMETER;

public final class GetDGSGood extends ApiServlet.JsonRequestHandler {

  private final ParameterService parameterService;

  public GetDGSGood(ParameterService parameterService) {
    super(new LegacyDocTag[] {LegacyDocTag.DGS}, GOODS_PARAMETER);
    this.parameterService = parameterService;
  }

  @Override
  protected
  JsonElement processRequest(HttpServletRequest req) throws BurstException {
    return JSONData.goods(parameterService.getGoods(req));
  }

}

File: src/main/kotlin/brs/api/http/GetDGSGood.kt
package brs.api.http

import brs.api.http.common.Parameters.GOODS_PARAMETER
import brs.services.ParameterService
import com.google.gson.JsonElement
import javax.servlet.http.HttpServletRequest

internal class GetDGSGood internal constructor(private val parameterService: ParameterService) :
    APIServlet.JsonRequestHandler(arrayOf(APITag.DGS), GOODS_PARAMETER) {

    override fun processRequest(request: HttpServletRequest): JsonElement {
        return JSONData.goods(parameterService.getGoods(request))
    }

}
